shap/plots/_beeswarm.py

```diff
@@ -415,21 +431,22 @@ def beeswarm(shap_values, max_display=10, order=Explanation.abs.mean(0),
 #         cb.ax.set_aspect((bbox.height - 0.9) * 20)
         # cb.draw_all()
 
-    pl.gca().xaxis.set_ticks_position('bottom')
-    pl.gca().yaxis.set_ticks_position('none')
-    pl.gca().spines['right'].set_visible(False)
-    pl.gca().spines['top'].set_visible(False)
-    pl.gca().spines['left'].set_visible(False)
-    pl.gca().tick_params(color=axis_color, labelcolor=axis_color)
-    pl.yticks(range(len(feature_inds)), reversed(yticklabels), fontsize=13)
-    pl.gca().tick_params('y', length=20, width=0.5, which='major')
-    pl.gca().tick_params('x', labelsize=11)
-    pl.ylim(-1, len(feature_inds))
-    pl.xlabel(labels['VALUE'], fontsize=13)
+    ax.xaxis.set_ticks_position('bottom')
+    ax.yaxis.set_ticks_position('none')
+    ax.spines['right'].set_visible(False)
+    ax.spines['top'].set_visible(False)
+    ax.spines['left'].set_visible(False)
+    ax.tick_params(color=axis_color, labelcolor=axis_color)
+    ax.set_yticks(range(len(feature_inds)), reversed(yticklabels), fontsize=13)
+    ax.tick_params('y', length=20, width=0.5, which='major')
+    ax.tick_params('x', labelsize=11)
+    ax.set_ylim(-1, len(feature_inds))
+    ax.set_xlabel(labels['VALUE'], fontsize=13)
     if show:
         pl.show()
     else:
-        return pl.gca()
+        pl.close(fig)
```

Comment:
May I ask about the motivation behind `pl.close`? I think it may be a breaking change; we may as well decide a suitable pattern and plan to do that same for the other plots.

Reply:
Of course. Jupyter will still render the plot even if `show = false`. Closing the figure will prevent that. But maybe I've misunderstood what show is meant to be doing.

Ah if its a breaking change then perhaps more thought should be put into it. How do you think it's a breaking change?

Reply:
I think this is "breaking" just in the sense that this is different behaviour than before. For now, let's avoid calling `plt.close` just to keep this plot consistent with the other plots in the library.

I think the "show" argument is really not needed and I'd like to deprecate it entirely, leaving it to the user to call `plt.show()` if desired. But, I think it would be better to make this change in one fell swoop, updating all the plots to a uniform interface.

Reply:
Note to self for future reference: It may be quite hard to "re-open" a plot that was closed with `plt.close()` , as described here:
https://stackoverflow.com/a/31731945

Reply:
I've created an issue to discuss this issue further: #3819

For now though, let's remove this line:

```suggestion
```